TRAF4.py

```diff
@@ -0,0 +1,176 @@
+            'g': 'c', 'G': 'C',
+            'c': 'g', 'C': 'G'
+        }
+        if 'T' in self.sequence.upper():
+            raise NuclAcidnucleotideError('T-contain sequence is not proper RNA sequence')
+    
+    def __repr__(self):
+        return f'RNASequence("{self.sequence}")'
+
+
+class AminoAcidSequence(BiologicalSequence):
+    def gravy(self):
+        """Calculate GRAVY (grand average of hydropathy) value"""
+        gravy_aa_values = {'L': 3.8,
+                           'K': -3.9,
+                           'M': 1.9,
+                           'F': 2.8,
+                           'P': -1.6,
+                           'S': -0.8,
+                           'T': -0.7,
+                           'W': -0.9,
+                           'Y': -1.3,
+                           'V': 4.2,
+                           'A': 1.8,
+                           'R': -4.5,
+                           'N': -3.5,
+                           'D': -3.5,
+                           'C': 2.5,
+                           'Q': -3.5,
+                           'E': -3.5,
+                           'G': -0.4,
+                           'H': -3.2,
+                           'I': 4.5}
+        gravy_aa_sum = 0
+        for amino_ac in self.sequence.upper():
+            gravy_aa_sum += gravy_aa_values[amino_ac]
+        return round(gravy_aa_sum / len(self.sequence), 3)
+    
+    def __repr__(self):
+        return f'AminoAcidSequence("{self.sequence}")'
+
+
+
+def make_thresholds(threshold: int | float | tuple) -> tuple:
+    """Check thresholds input and convert single value to tuple"""
+    if isinstance(threshold, int) or isinstance(threshold, float):
+        lower = 0
+        upper = threshold
+    else:
+        lower = threshold[0]
+        upper = threshold[1]
+    return lower, upper
+
+
+def filter_fastq(input_path: str,
+                 gc_thresholds: int | float | tuple = (20, 80),
+                 len_thresholds: int | float | tuple = (0, 2 ** 32),
+                 quality_threshold: int | float = 0,
+                 output_path: str = 'filtered.fastq'):
+    """Filters out sequences from fastq file by the specified conditions:
+        - GC-content, inside interval include borders, or, if single value, not bigger than specified
+        - length, inside interval include borders, or, if single value, not bigger than specified
+        - average phred scores, not less than specified
+        Default output file name 'filtered.fastq'
+    """
+    records = list(SeqIO.parse(input_path, 'fastq'))
+
+    filtered_1_gc_idxs = []
+    filtered_2_len_idxs = []
+    filtered_3_phred_idxs = []
+    filtered_results = []
+
+    min_gc, max_gc = make_thresholds(gc_thresholds)
+    min_len, max_len = make_thresholds(len_thresholds)
+
+    for count, record in enumerate(records):
+        gc_percent = gc_fraction(record.seq) * 100
+        if min_gc <= gc_percent <= max_gc:
+            filtered_1_gc_idxs.append(count)
```

Comment:
Можно использовать более читаемую конструкцию for record in records вместо for count, record in enumerate(records).

```suggestion
    for record in records:
    gc_percent = gc_fraction(record.seq) * 100
    if min_gc <= gc_percent <= max_gc:
        filtered_1_gc_idxs.append(record)
```